modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]DefineAPISchemasWithinAnAPIGroup/EditAPIApplicationSchema.testcase

```diff
@@ -0,0 +1,216 @@
+		property portal.acceptance = "true";
+
+		task ("And Given create an published API application with related schema") {
+			APIBuilderUI.createAPISchema(
+				confirmCreation = "true",
+				fieldDescription = "Description default",
+				name = "testSchema",
+				objectDefinitionName = "APIApplication");
+
+			Alert.viewSuccessMessage();
+
+			Click(locator1 = "APIBuilder#PUBLISH_BUTTON");
+		}
+
+		task ("And Given edit the schema in Edit API Application > Schemas") {
+			Type(
+				locator1 = "TextInput#NAME",
+				value1 = "testSchema edited");
+
+			Type(
+				locator1 = "TextInput#DESCRIPTION_FIELD",
+				value1 = "Description edited");
+		}
+
+		task ("When I save changed name and description") {
+			Click.pauseClickAt(locator1 = "APIBuilder#PUBLISH_BUTTON");
+		}
+
+		task ("Then I can see success alert message about API schema changes were saved") {
+			Alert.viewSuccessMessage();
+		}
+
+		task ("And Then I can see updated name and description in Info tab") {
+			AssertElementPresent(
+				locator1 = "TextInput#NAME",
+				value1 = "testSchema edited");
+
+			AssertElementPresent(
+				locator1 = "TextInput#DESCRIPTION_FIELD",
+				value1 = "Description edited");
+		}
+	}
+
+	@priority = 3
+	test CanGoBackToSchemasListWithCancelEdit {
+		property portal.acceptance = "true";
+
+		task ("And Given create an published API application with related schema") {
+			APIBuilderUI.createAPISchema(
+				confirmCreation = "true",
+				fieldDescription = "Description default",
+				name = "testSchema",
+				objectDefinitionName = "APIApplication");
+
+			Alert.viewSuccessMessage();
+
+			Click(locator1 = "APIBuilder#PUBLISH_BUTTON");
+		}
+
+		task ("And Given edit the schema in Edit API Application > Schemas") {
+			Type(
+				locator1 = "TextInput#NAME",
+				value1 = "testSchema edited");
+
+			Type(
+				locator1 = "TextInput#DESCRIPTION_FIELD",
+				value1 = "Description edited");
+		}
+
+		task ("When I click Cancel button without any changes") {
+			Button.clickCancel();
+
+			Click(locator1 = "APIBuilder#CONTINUE_WITHOUT_SAVING_BUTTON");
+		}
+
+		task ("Then I can see the schema under Edit API Application > Schemas") {
+			AssertElementPresent(
+				key_name = "testSchema",
+				locator1 = "ObjectAdmin#TABLE_LIST_TITLE");
+
+			AssertElementPresent(
+				key_tabName = "Description default",
+				locator1 = "ObjectAdmin#ANY_TAB_ON_ENTRY");
+		}
+	}
+
+	@priority = 4
+	test CannotEditSchemaRelatedObject {
+		property portal.acceptance = "true";
+
+		task ("And Given create an published API application with related schema") {
+			APIBuilderUI.createAPISchema(
+				confirmCreation = "true",
+				fieldDescription = "Description default",
+				name = "testSchema",
+				objectDefinitionName = "APIApplication");
+
+			Alert.viewSuccessMessage();
+
+			Click(locator1 = "APIBuilder#PUBLISH_BUTTON");
+		}
+
+		task ("And Given edit the schema in Edit API Application > Schemas") {
+			Type(
```

Comment:
Just a note that we can create a new macro to edit schema name and description in APIBuilderUI.macro if you'd like to :smile: